### PR TITLE
Add netcore50 configuration for System.Net.Http.Rtc

### PR DESCRIPTION
**Pull request by mellinoe**
This contract is only supported on .NET Native.

NOTE: This contract type-forwards its only type to System.Net.Http, the implementation assembly. I have ported this as-is from our internal build, but we probably want to change this. The type, `RtcRequestFactory` is a static class with only one static method on it, so I don't see why it can't just be implemented in System.Net.Http.Rtc itself. At the very least, we can share the same source file with System.Net.Http if it still needs to live there. @davidsh Do you recall why these two assemblies were factored as such?

@weshaggard , @CIPop

### CODE REVIEW

**Comment by davidsh** [comment]
cc: @ericstj

**Comment by davidsh** [comment]
`RtcRequestFactory` is currently in the System.Net.Http binary because that is how it is implemented on .NET Framework (Desktop).  @ericstj advised us to put the `RtcRequestFactory` type into System.Net.Http because, at the time, he didn't want to have a separate NuGet package for it.

We could probably put it into a separate package but there a bunch of TFS things that would need changing if we are to add a new NuGet package (with only netcore50 target) to the build system.

I'm fine if we want to do that.  It's just more work.

**Comment by davidsh** [comment]
Actually, in looking at this more, we already have a separate System.Net.Http.Rtc NuGet package.  Currently, it is just a facade to System.Net.Http.

So, if you want to make the TFS side changes as well, you can move the entire type to the System.Net.Http.Rtc binary and make it a pure implementation dll instead of facade. Except you'll need to do the net46 (.NET Framework Desktop) facade work as well.

**Comment by mellinoe** [comment]
I don't think this contract is supported on .NET Framework, it only seems to be supported on .NET Native. Additionally, I'm not sure that `RtcRequestFactory` is even used in System.Net.Http. I think we can just move the source file into this project.

**Comment by davidsh** [comment]
`RtcRequestFactory` was implemented in the .NET Framework (Desktop) primarly to support Windows Store Apps for Windows 8.0 and 8.1.  The implementation itself lives in the .NET Framework (Desktop) System.dll binary.  When contracts were first implemented, i.e. the original Windows Store Profile for .NET apps, the System.Net.Http.Rtc "contract" was born.

When UWP and Windows 10 happened, the HTTP stack implemented changed and moved on top of the WinRT Windows.Web.Http APIs.  This caused the architecture of `RtcRequestFactory` to not be possible anymore.  Essentially at this point, this type, which wasn't used much, has become deprecated unofficially.

To preserve source re-compiling compat, we keep the type and this ensures that the tooling (compilers, etc) still work.  It was decided by @terrajobst  and @ericstj that the best solution was to keep the type and just throw PlatformNotSupportedException from the static method.

**Comment by mellinoe** [comment]
Thanks, that's some good background. I'm pretty sure there's nothing stopping us from moving the type into the System.Net.Http.Rtc implementation assembly itself, so I updated my commit to move the source file into the project.

**Comment by mellinoe** [comment]
Innerloop Windows_NT Debug Build and Test

(Socket test timeout)

File: src/System.Net.Http.Rtc/src/System.Net.Http.Rtc.csproj [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  <PropertyGroup>
+    <Configuration Condition="'$(Configuration)'==''">netcore50_Debug</Configuration>
+  </PropertyGroup>
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <PropertyGroup>
+    <AssemblyName>System.Net.Http.Rtc</AssemblyName>
+    <AssemblyVersion>4.0.1.0</AssemblyVersion>
+  </PropertyGroup>
+  <!-- Help VS understand available configurations -->
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcore50_Debug|AnyCPU'" />
```
Comment (weshaggard) [inline src/System.Net.Http.Rtc/src/System.Net.Http.Rtc.csproj]: While this is technically IL only it is very much Windows specific and I would prefer we not having it be build in the other configurations so can you update the various configurations here and in the builds file to make it OSGroup=Windows.

**Comment by weshaggard** [comment]
LGTM. I'm good with putting the dummy implementation into this assembly.

File: src/System.Net.Http.Rtc/src/System.Net.Http.Rtc.csproj (hunk 1 above)
Comment (mellinoe) [inline src/System.Net.Http.Rtc/src/System.Net.Http.Rtc.csproj]: Done

**Comment by mellinoe** [comment]
CI will re-run now, I pushed the change you suggested (OSGroup).

**Comment by mellinoe** [comment]
Test Innerloop Windows_NT Debug Build and Test

**Comment by davidsh** [comment]
LGTM

**Pull request merged**